src/persistence/settingsserializer.cpp

```diff
@@ -180,9 +188,12 @@ const SettingsSerializer::Value* SettingsSerializer::findValue(const QString& ke
         {
             if (a.group != group)
                 continue;
-            for (int vi : a.values)
-                if (values[vi].arrayIndex == arrayIndex && values[vi].key == key)
-                    return &values[vi];
+            for (int vi = 0; vi < a.values.size(); vi++)
+            {
+                const Value& v = values[vi];
+                if (v.arrayIndex == arrayIndex && v.key == key)
+                    return &v;
+            }
```

Comment:
@tux3 Can you check this again please. Should be good now! Sorry, I got confused by the array index and the contained index value. Thanks for spotting this :innocent:. However, I now corrected the types in header, so most of the casts completely vanished.